assets/js/blocks/product-query/variations/products-on-sale.tsx

```diff
@@ -43,7 +43,7 @@ if ( isExperimentalBuild() ) {
 			[
 				'core/post-template',
 				{},
-				[ [ 'core/post-title' ], [ 'core/post-featured-image' ] ],
+				[ [ 'core/post-title' ], [ 'woocommerce/product-image' ] ],
```

Comment:
Same for the on sale variation.
```suggestion
				[ [ 'woocommerce/product-image' ], [ 'core/post-title' ] ],
```